mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
well even if takeLast() doesn't delete, I still think there might be a minor problem with append, since even though Qt says "QList preallocates extra space on both sides of its internal buffer to allow for fast growth at both ends of the list" I still think it has to dynamically allocate space if use up too much.  But I'm have a feeling this is not the performance bug I'm searching for...